lib/logstash/runner.rb

```diff
@@ -3,22 +3,72 @@
 Encoding.default_external = Encoding::UTF_8
 $DEBUGLIST = (ENV["DEBUG"] || "").split(",")
 
+require "clamp" # gem 'clamp'
+require "net/http"
 require "logstash/environment"
 
 LogStash::Environment.load_locale!
 
 require "logstash/namespace"
-require "logstash/program"
+require "logstash/agent"
 
-class LogStash::Runner
-  include LogStash::Program
+class LogStash::Runner < Clamp::Command
 
-  def main(args)
+  DEFAULT_INPUT = "input { stdin { type => stdin } }"
+  DEFAULT_OUTPUT = "output { stdout { codec => rubydebug } }"
+
+  option ["-f", "--config"], "CONFIG_PATH",
+    I18n.t("logstash.runner.flag.config"),
+    :attribute_name => :config_path
+
+  option "-e", "CONFIG_STRING",
+    I18n.t("logstash.runner.flag.config-string",
+           :default_input => DEFAULT_INPUT, :default_output => DEFAULT_OUTPUT),
+    :default => "", :attribute_name => :config_string
+
+  option ["-w", "--filterworkers"], "COUNT",
+    I18n.t("logstash.runner.flag.filterworkers"),
+    :attribute_name => :filter_workers, :default => 1, &:to_i
+
+  option ["-l", "--log"], "FILE",
+    I18n.t("logstash.runner.flag.log"),
+    :attribute_name => :log_file
+
+  # Old support for the '-v' flag'
+  option "-v", :flag,
```

Comment:
I'm open to that, though it may be surprising to users.  If anything, I'd prefer that `-v` went away and did _not_ change to report the version - mostly to avoid surprising users.